src/Admin/PageAdmin.php

```diff
@@ -196,7 +198,9 @@ protected function configureDatagridFilters(DatagridMapper $filter): void
             ->add('parent')
             ->add('edited')
             ->add('hybrid', CallbackFilter::class, [
-                'callback' => static function ($queryBuilder, $alias, $field, $data): void {
+                'callback' => static function (ProxyQueryInterface $queryBuilder, string $alias, string $field, array $data): void {
+                    \assert($queryBuilder instanceof ProxyQuery);
+
                     if (\in_array($data['value'], ['hybrid', 'cms'], true)) {
                         $queryBuilder->andWhere(sprintf('%s.routeName %s :routeName', $alias, 'cms' === $data['value'] ? '=' : '!='));
                         $queryBuilder->setParameter('routeName', PageInterface::PAGE_ROUTE_CMS_NAME);
```

Comment:
Might be better to rely on the getQueryBuilder() method which exists on the two interface if we want to later add mongo support https://github.com/sonata-project/SonataDoctrineMongoDBAdminBundle/blob/4.x/src/Datagrid/ProxyQueryInterface.php

Reply:
Oh, ok I will try.

Reply:
Fixed.